File: MyFinance.Application/Dtos/TransactionResponseDto.cs
using MyFinance.Domain.Enums;
using System;

namespace MyFinance.Application.Dtos;

public class TransactionResponseDto
{
    public Guid Id { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public TransactionType Type { get; set; }
    public string TypeName { get; set; } = string.Empty;
    public DateTime Date { get; set; }
    public DateTime CreatedAt { get; set; }
    public Guid AccountId { get; set; }
    public string AccountName { get; set; } = string.Empty;
    public Guid CategoryId { get; set; }
    public string CategoryName { get; set; } = string.Empty;
}

File: MyFinance.Frontend/src/types/TransactionResponseDto.ts
import type { TransactionType } from './TransactionType';

export interface TransactionResponseDto {
    id: string;
    description: string;
    amount: number;
    type: TransactionType;
    typeName: string; // "Income" ou "Expense"
    date: string;
    createdAt: string;
    accountId: string;
    accountName: string;
    categoryId: string;
    categoryName: string;
}